### ISSUE TITLE
[🐞] State Not Refreshing/Updating Inside Form Action Callback

### ISSUE THREAD

**Issue by nla-asia** [BuilderIO/qwik#4853]
### Which component is affected?

Qwik Runtime

### Describe the bug

I want to show a loading indicator when the form is submitted using routeAction as follows but it seems the state is not updating even though the callback gets called after the submission is completed. 

```typescript
export default component$(() => {

  const action = useFormSubmitAction();
  const processing = useSignal(0);
 
  return (
      <article class="glass">
            <Form 
            action={action}
            onSubmit$={()=>{
              processing.value = 1;
            }}
            onSubmitCompleted$={(d)=>{
                console.log("submission done");
              processing.value = 0;
            }}
            spaReset
            >
             <div class="flex">

                    ....
                
                    <button type="submit" class="btn">{processing.value==1? 'Posting...':'Post'}</button>
                
              </div>
            </Form>
      </article>
  );
});
```

### Reproduction

https://qwik.new/

### Steps to reproduce

Just run the route component posted above.

### System Info

```shell
Windows 11 Nodejs 20
```

**Comment by kikuomax** [comment]
@nla-asia I faced a similar problem. In my case, `onSubmit$` was somehow notified after `onSubmitCompleted$`. I found `action` had `isRunning` property and I worked around the problem by using `action.isRunning` instead of a flag controlled by `onSubmit$` and `onSubmitCompleted$` events. I do not think `isRunning` is documented, so I do not know we can rely on it, though.
https://github.com/BuilderIO/qwik/blob/ab74880eeaaed4a247f369ce6754ddcd966176b7/packages/qwik-city/runtime/src/server-functions.ts#L105
